Xamarin.Forms.Platform.Android/CollectionView/ItemsSourceFactory.cs

```diff
@@ -21,8 +23,8 @@ public static IItemsViewSource Create(IEnumerable itemsSource, ICollectionChange
 
 			switch (itemsSource)
 			{
-				case IList _ when itemsSource is INotifyCollectionChanged:
-					return new ObservableItemsSource(itemsSource as IList, notifier);
+				case IList list when itemsSource is INotifyCollectionChanged:
+					return new ObservableItemsSource(new MarshalingObservableCollection(list), notifier);
```

Comment:
What about iOS? Would it be necessary to use `MarshalingObservableCollection` in iOS too?